lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -1928,6 +1928,61 @@ LogicalResult ConvertAtenOp<ValueTensorLiteralOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenFlattenUsingIntsOp>::matchAndRewrite(
+    AtenFlattenUsingIntsOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  // Not a ranked tensor type
+  auto selfType = adaptor.self().getType().dyn_cast<RankedTensorType>();
+  if (!selfType || !selfType.hasStaticShape())
+    return op.emitError(
+        "Only ranked tensor types with static shapes are currently supported");
+
+  int64_t start_dim, end_dim;
+
+  if (!matchPattern(op.start_dim(), m_TorchConstantInt(&start_dim)))
+    return op.emitError("start_dim must be a Scalar constant");
+
+  if (!matchPattern(op.end_dim(), m_TorchConstantInt(&end_dim)))
+    return op.emitError("end_dim must be a Scalar constant");
+
+  if (end_dim != -1 && end_dim < start_dim)
+    return op.emitError("end_dim must be larger than start_dim");
+
+  SmallVector<int64_t> newShape;
+  uint32_t idx = 0;
+  bool acc = false;
+  for (auto s : selfType.getShape()) {
```

Comment:
nit: could use llvm::enumerate() for this, then the var `idx` would not be needed. see example here https://github.com/llvm/mlir-npcomp/blob/52ed3313b49b8f56a6dac270ee334c6dffebcaf2/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L2886